packages/keyring-api/src/sol/types.ts

```diff
@@ -0,0 +1,44 @@
+import type { Infer } from '@metamask/superstruct';
+import { array, enums, literal } from '@metamask/superstruct';
+
+import { KeyringAccountStruct, SolAccountType } from '../api';
+import { object, definePattern } from '../superstruct';
+
+/**
+ * Solana addresses are encoded using Base58
+ * Represented as 32 bytes in the format of an Ed25519 PublicKey
+ * And a valid Solana address is 44 characters long
```

Comment:
It doesn't match the regex, did you mean it can be between 32 and 44 characters long?

Reply:
yes, they are normally 44 but can also be 32

Reply:
I'll edit the comment 👍🏼